core/integration/module_typescript_test.go

```diff
@@ -641,6 +641,140 @@ func (ModuleSuite) TestTypescriptRuntimeDetection(ctx context.Context, t *testct
 	})
 }
 
+func (ModuleSuite) TestTypeScriptPackageManagerDetection(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	t.Run("should default to yarn", func(ctx context.Context, t *testctx.T) {
+		modGen := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work").
+			With(daggerExec("init", "--name=Package-Detection", "--sdk=typescript", "--source=."))
+
+		files, err := modGen.Directory(".").Entries(ctx)
+		require.NoError(t, err)
+		require.Contains(t, files, "yarn.lock")
+
+		// Check that the package manager is set to yarn.
+		packageManager, err := modGen.File("package.json").Contents(ctx)
+		require.NoError(t, err)
+		require.Contains(t, packageManager, `"packageManager": "yarn@1.22.22`)
```

Comment:
By using corepack for yarn (because upper version of yarn [needs corepack](https://yarnpkg.com/corepack)), it automatically set the field so I'm not sure it's a good idea to remove it.
But the user is free to override it after that

Reply:
What I mean is that the runtime module shouldn't generate a default config in `package.json`. Basically if it's not set then use the default, rather then writing it to the user's `package.json`.

Reply:
corepack automatically do it so I'm not sure how I can do that.

Reply:
I thought it was a dagger specific field, under `"dagger"`, like the runtime setting.